### ISSUE TITLE
Suggestions for the STAT545 website

### ISSUE THREAD

**Issue by jennybc** [STAT545-UBC/Discussion#84]
STAT 545 runs again September - December 2015. I don't anticipate the __major__ content creation / renovation that has been necessary in the past two cycles.

I do want to up our game in terms of the website.

I'd appreciate __constructive criticism__ from anyone who's ever used the site!

Specific hopes, dreams, and nightmares I already have:

  * Current file naming & organization scheme, e.g. "block009_dplyr-intro.rmd", probably not sustainable
    - renumbering or putting things into folders would break links
    - number of files sitting in one directory has gotten out of hand
    - I need to revisit rmarkdown/knitr and see if recent promised improvements re: flexibility around working directory and directories for intermediate and output files really work for me
  * Need better CSS / styling. For example, pages feel like wall of text and text runs across too great a width.
  * How hard to get content ready for some e-Book system, once we're touching all of it anyway?
  * We need a better way to find info than the [topics](http://stat545-ubc.github.io/topics.html) page.
  * Simplicity and (lean) dependencies are very important to me. There's a reason I'm NOT using jekyll for this already. So many gems. Such updating. Wow.

@sjackman @daattali @aammd @jooolia @gloriali

**Comment by sjackman** [comment]
> How hard to get content ready for some e-Book system, once we're touching all of it anyway?

I don't know anything about it, but [GitBook](https://www.gitbook.com/) looks interesting.

**Comment by sjackman** [comment]
> Simplicity and (lean) dependencies are very important to me. There's a reason I'm NOT using jekyll for this already. So many gems. Such updating. Wow.

I would consider [Using Jekyll with Pages](https://help.github.com/articles/using-jekyll-with-pages/). `gem install github-pages` works pretty well.

**Comment by sjackman** [comment]
> Need better CSS / styling.

I know not much about CSS, but I know that [Bootstrap](http://getbootstrap.com/) is a popular thing.

**Comment by daattali** [comment]
re: css/styling: I think one trend that I see a lot nowadays is to have lots of whitespace, bigger fonts, and restrict the width of the page. Most of the content *is* just plain text/code without images so I'm not sure if just adding those elements will make a difference.

You be the judge whether it's more readable/less intimidating or not:  [current version](http://stat545-ubc.github.io/block000_r-rstudio-install.html) vs [more space, bigger font](http://deanattali.com/stat545test)

**Comment by sjackman** [comment]
:+1: for bigger font and skinnier columns of text

**Comment by jennybc** [comment]
From [Sharon Machlis](https://twitter.com/sharon000/status/629427280939479040) via twitter:

> Online UI nit: It's not 100% clear from home pg all info available. Suggest a Dig into Details LINK where More info? is

**Comment by sjackman** [comment]
Another option for a GitHub eBook. https://leanpub.com/

**Comment by daattali** [comment]
Just dumping this here: GitHub has just released a new tool GitHub Desktop for Windows+Mac users for interacting with github. Seems like it's similar to SourceTree. I haven't tried it out but I remember that last year we gave the students several different options for what GUI software to use with git and a few of them gave us lots of headaches, so maybe it'd be worth taking a look, although SourceTree seems to do the job
 
https://github.com/blog/2046-github-desktop-is-now-available

**Comment by sjackman** [comment]
I'll mention here that I also really like [GitUp](http://gitup.co/) for Mac, particularly for the task of git repository surgery.

**Comment by jennybc** [comment]
This blog has a nice page width and font size -- worth copying/emulating!

http://www.perfectlyrandom.org/2015/06/16/never-trust-the-row-names-of-a-dataframe-in-R/

**Comment by jennybc** [comment]
As @daattali just pointed out in https://github.com/STAT545-UBC/STAT545-UBC.github.io/issues/20, backend of website could be more automated. Listing this here in case a TA wants to work on such things. Example:

  * automate the knitting locally (i.e. help Jenny w/ her awful unfinished bulk SIMPLE knitting package or write the damn Makefile we need)
  * automate the knitting elsewhere with some fancy CI or Git hook stuff
  * is there some reasonable notion of "testing" the website build after someone makes an edit? how can we harness that to make it easier to vet and accept change from, e.g., students?
  * add "edit this page" functionality

**Comment by daattali** [comment]
Another suggestion: buy a domain eg. http://stat545.com or anything else, mostly just because current URL is difficult for most new people to remember ("what is this github.io business?!"). Shouldn't be more than $15/year, so it's only a question of whether you want that or not

**Comment by jennybc** [comment]
@daattali Great idea! Where should I buy/register? I've only done that once before and I wondered if it mattered who I used.

**Comment by daattali** [comment]
I'm no expert, I've only done very minimal research into this before buying my own website earlier this year, but from what I gather using any big company will be just fine and not make any difference. I personally used http://namecheap.com

**Comment by daattali** [comment]
Small issue: In the [FAQ page](http://stat545-ubc.github.io/faq.html) under "Do I need a computer?" the answer makes it sound like a computer is necessary for class meetings.  "Students will get the most out of this if they can bring their own laptop to class every day. If this is not possible, we will try to help you work something out."

This was true last year, but it's my understanding that this year we have a room equipped with computers so maybe that could be clarified

**Comment by daattali** [comment]
Nevermind, looks like the room does not have computers.  Also looks like I arrived half an hour too early :/

**Comment by daattali** [comment]
Regarding new theme/css: New CSS has been implemented, previous theme was discarded. But as of 2015-09-15 only a few pages are using the new CSS. All pages need to be re-rendered to get updated.

**Comment by daattali** [comment]
Another suggestion that's super easy to implement: add Google Analytics. It'd be nice to be able to see how much traffic the site is getting, which pages are popular, where referrals are coming from (twitter? google searches? direct urls?), what countries people are coming from.  Especially since we do advertise the site sometimes to many people outside of the class.  I personally link to the course website on reddit and stackoverflow sometimes when it can be relevant, and I wonder how much visibility that gives us

**Comment by jennybc** [comment]
Yes. I do that on some other (non-GitHub Pages) stuff but apparently neglected to do here. Can you make that so? You are an owner....

**Comment by daattali** [comment]
Done. But only new pages that will be rendered will have the googleanalytics script in them.

**Comment by jennybc** [comment]
Thanks, @daattali. Yes I know we need to re-render everything.
